### PR TITLE
[WIP][TM-256] Spring Boot viewer-ng module

### PR DESCRIPTION
**Pull request by mprins**
- [x] setup maven module
- [ ] port `retrieveAppConfigJSON`
- [ ] ...

close TM-256

### CODE REVIEW

File: viewer-ng/pom.xml [hunk 1]
```diff
@@ -0,0 +1,115 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <!-- would have org.springframework.bootspring-boot-starter-parent</artifactId> as parent, but we need TM as a parent -->
```
Comment (mtoonen) [inline viewer-ng/pom.xml]: Possibly dumb question: using the springboot starter as a parent in the tailormap parent module? Maybe for now not possible (everything would get the springboot deps)

File: viewer-ng/src/test/java/nl/tailormap/viewer_ng/HSQLDBTestProfileJPAConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        LocalContainerEntityManagerFactoryBean em = new LocalContainerEntityManagerFactoryBean();
+        em.setPersistenceUnitName("viewer-config-hsqldb");
+        em.setDataSource(dataSource());
+
+        return em;
+    }
+
+    @Bean
+    @Profile("test")
+    public DataSource dataSource() {
```
Comment (mtoonen) [inline viewer-ng/src/test/java/nl/tailormap/viewer_ng/HSQLDBTestProfileJPAConfiguration.java:47]: This is not necessary when having a separate test propertiesfile (which you have) and you add a annotation to the test-class: 
@TestPropertySource(
        locations = "classpath:application-integrationtest.properties").

When using the default propertynames + hsqldb dependency in you pom file, springboot will automatically create the datasource.

File: viewer-ng/src/main/webapp/META-INF/context.xml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<Context antiResourceLocking="false" path="/viewer">
```
Comment (mtoonen) [inline viewer-ng/src/main/webapp/META-INF/context.xml]: Nice, I couldn't get this to work...

**Review by mtoonen — APPROVED** [review]
Some small changes, but overall looks promising

File: viewer-ng/pom.xml (hunk 1 above)
Comment (mprins) [reply viewer-ng/pom.xml]: springboot starte doesn't really do anything apart from 1 plugin; we're using the upstream parent as a BOM in our parent now, so we get all the right versions, but can still override with a latest version eg. Hibernate

File: viewer-ng/src/test/java/nl/tailormap/viewer_ng/HSQLDBTestProfileJPAConfiguration.java (hunk 2 above)
Comment (mprins) [reply viewer-ng/src/test/java/nl/tailormap/viewer_ng/HSQLDBTestProfileJPAConfiguration.java:47]: yeah need to figure that out, needed to upgrade to JUnit 5 first (#2827 )

Comment (mprins) [reply viewer-ng/src/test/java/nl/tailormap/viewer_ng/HSQLDBTestProfileJPAConfiguration.java:47]: it turns out I need to override some things that are specified in the persistence.xml; I have something working now that (for each test):

1. creates a database with a unique name
2. loads the test data

so integration/unit tests can run multithreaded

**Comment by mprins** [comment]
moved to https://github.com/B3Partners/tailormap-api

**Pull request closed without merging**